File: app-miyou/src/main/java/com/miyou/app/application/credit/port/CreditTransactionRepository.java
package com.miyou.app.application.credit.port;

import org.springframework.data.domain.Pageable;

import com.miyou.app.domain.credit.model.CreditTransaction;
import com.miyou.app.domain.dialogue.model.UserId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CreditTransactionRepository {

	Mono<CreditTransaction> save(CreditTransaction transaction);

	Flux<CreditTransaction> findByUserIdOrderByCreatedAtDesc(UserId userId, Pageable pageable);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/credit/port/CreditTransactionRepository.kt
package com.miyou.app.application.credit.port

import com.miyou.app.domain.credit.model.CreditTransaction
import com.miyou.app.domain.dialogue.model.UserId
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface CreditTransactionRepository {
    fun save(transaction: CreditTransaction): Mono<CreditTransaction>

    fun findByUserIdOrderByCreatedAtDesc(
        userId: UserId,
        pageable: Pageable,
    ): Flux<CreditTransaction>
}
